Mathlib/Analysis/NormedSpace/Multilinear/Basic.lean

```diff
@@ -1286,3 +1218,162 @@ instance continuousConstSMul : ContinuousConstSMul R (ContinuousMultilinearMap 
+  have cau : ∀ v, CauchySeq fun n => f n v := by
+    intro v
+    apply cauchySeq_iff_le_tendsto_0.2 ⟨fun n => b n * ∏ i, ‖v i‖, _, _, _⟩
+    · intro
+      exact mul_nonneg (b0 _) (nonneg v)
+    · intro n m N hn hm
+      rw [dist_eq_norm]
+      apply le_trans ((f n - f m).le_op_norm v) _
+      exact mul_le_mul_of_nonneg_right (b_bound n m N hn hm) (nonneg v)
+    · simpa using b_lim.mul tendsto_const_nhds
+  -- We assemble the limits points of those Cauchy sequences
+  -- (which exist as `G` is complete)
+  -- into a function which we call `F`.
+  choose F hF using fun v => cauchySeq_tendsto_of_complete (cau v)
+  -- Next, we show that this `F` is multilinear,
+  let Fmult : MultilinearMap 𝕜 E G :=
+    { toFun := F
+      map_add' := fun v i x y => by
+        skip
+        have A := hF (Function.update v i (x + y))
+        have B := (hF (Function.update v i x)).add (hF (Function.update v i y))
+        simp? at A B says simp only [map_add] at A B
+        exact tendsto_nhds_unique A B
+      map_smul' := fun v i c x => by
+        skip
+        have A := hF (Function.update v i (c • x))
+        have B := Filter.Tendsto.smul (@tendsto_const_nhds _ ℕ _ c _) (hF (Function.update v i x))
+        simp? at A B says simp only [map_smul] at A B
+        exact tendsto_nhds_unique A B }
+  -- and that `F` has norm at most `(b 0 + ‖f 0‖)`.
+  have Fnorm : ∀ v, ‖F v‖ ≤ (b 0 + ‖f 0‖) * ∏ i, ‖v i‖ := by
+    intro v
+    have A : ∀ n, ‖f n v‖ ≤ (b 0 + ‖f 0‖) * ∏ i, ‖v i‖ := by
+      intro n
+      apply le_trans ((f n).le_op_norm _) _
+      apply mul_le_mul_of_nonneg_right _ (nonneg v)
+      calc
+        ‖f n‖ = ‖f n - f 0 + f 0‖ := by
+          congr 1
+          abel
+        _ ≤ ‖f n - f 0‖ + ‖f 0‖ := (norm_add_le _ _)
+        _ ≤ b 0 + ‖f 0‖ := by
+          apply add_le_add_right
+          simpa [dist_eq_norm] using b_bound n 0 0 (zero_le _) (zero_le _)
+    exact le_of_tendsto (hF v).norm (eventually_of_forall A)
+  -- Thus `F` is continuous, and we propose that as the limit point of our original Cauchy sequence.
+  let Fcont := Fmult.mkContinuous _ Fnorm
+  use Fcont
+  -- Our last task is to establish convergence to `F` in norm.
+  have : ∀ n, ‖f n - Fcont‖ ≤ b n := by
+    intro n
+    apply op_norm_le_bound _ (b0 n) fun v => ?_
+    have A : ∀ᶠ m in atTop, ‖(f n - f m) v‖ ≤ b n * ∏ i, ‖v i‖ := by
+      refine' eventually_atTop.2 ⟨n, fun m hm => _⟩
+      apply le_trans ((f n - f m).le_op_norm _) _
+      exact mul_le_mul_of_nonneg_right (b_bound n m n le_rfl hm) (nonneg v)
+    have B : Tendsto (fun m => ‖(f n - f m) v‖) atTop (𝓝 ‖(f n - Fcont) v‖) :=
+      Tendsto.norm (tendsto_const_nhds.sub (hF v))
+    exact le_of_tendsto B A
+  rw [tendsto_iff_norm_sub_tendsto_zero]
+  exact squeeze_zero (fun n => norm_nonneg _) this b_lim
+
+end ContinuousMultilinearMap
+
+end Norm
+
+section Norm
+
+/-! Results that are only true if the source and target spaces are `NormedAddCommGroup`s
+(and not just `SeminormedAddCommGroup`s).-/
```

Comment:
Please fix this comment: the codomain doesn't need to be Hausdorff.

Reply:
Fixed.